### PR TITLE
[Merged by Bors] - Openshift compatibility

### PR DESCRIPTION
**Pull request by adwk67**
# Description

Add RBAC components for the operator to run on Openshift.

Subset of tests on openshift:

image

Subset of tests on openshift II:
```
--- PASS: kuttl (2095.61s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/logging_zookeeper-latest-3.8.0-stackable23.4.0-rc2_nifi-1.18.0-stackable23.4.0-rc2 (134.70s)
        --- PASS: kuttl/harness/cluster_operation_zookeeper-latest-3.8.0-stackable23.4.0-rc2_nifi-latest-1.18.0-stackable23.4.0-rc2 (254.16s)
        --- PASS: kuttl/harness/orphaned_resources_zookeeper-latest-3.8.0-stackable23.4.0-rc2_nifi-1.18.0-stackable23.4.0-rc2 (356.38s)
        --- PASS: kuttl/harness/upgrade_zookeeper-latest-3.8.0-stackable23.4.0-rc2_nifi_old-1.16.3-stackable23.4.0-rc2_nifi_new-1.18.0-stackable23.4.0-rc2 (379.31s)
        --- PASS: kuttl/harness/resources_zookeeper-latest-3.8.0-stackable23.4.0-rc2_nifi-1.18.0-stackable23.4.0-rc2 (41.03s)
        --- PASS: kuttl/harness/ldap_zookeeper-latest-3.8.0-stackable23.4.0-rc2_nifi-1.18.0-stackable23.4.0-rc2_ldap-use-tls-false_openshift-true (219.23s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.8.0-stackable23.4.0-rc2_nifi-1.18.0-stackable23.4.0-rc2_listener-class-external-unstable (253.14s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.8.0-stackable23.4.0-rc2_nifi-1.18.0-stackable23.4.0-rc2_listener-class-cluster-internal (239.95s)
        --- PASS: kuttl/harness/ldap_zookeeper-latest-3.8.0-stackable23.4.0-rc2_nifi-1.18.0-stackable23.4.0-rc2_ldap-use-tls-true_openshift-true (207.32s)
```

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by maltesander — CHANGES_REQUESTED** [review]
Just minor things.

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -91,6 +91,9 @@ pub struct NifiSpec {
     #[serde(default, skip_serializing_if = "Option::is_none")]
     /// Available NiFi roles
     pub nodes: Option<Role<NifiConfigFragment>>,
+    /// Emergency stop button, if `true` then all pods are stopped without affecting configuration (as setting `replicas` to `0` would)
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub stopped: Option<bool>,
```
[the remark this answers is not in the document]
Comment (adwk67) [reply rust/crd/src/lib.rs]: Done.

File: rust/operator-binary/src/controller.rs [hunk 2]
```diff
@@ -9,13 +9,15 @@ use std::{
 
 use rand::{distributions::Alphanumeric, Rng};
 use snafu::{OptionExt, ResultExt, Snafu};
+use stackable_operator::cluster_resources::ClusterResourceApplyStrategy;
+use stackable_operator::commons::rbac::build_rbac_resources;
```
[the remark this answers is not in the document]
Comment (adwk67) [reply rust/operator-binary/src/controller.rs]: Done.

File: rust/crd/src/lib.rs [hunk 3]
```diff
@@ -7,10 +7,10 @@ use std::collections::BTreeMap;
 use affinity::get_affinity;
 use serde::{Deserialize, Serialize};
 use snafu::{OptionExt, ResultExt, Snafu};
+use stackable_operator::commons::cluster_operation::ClusterOperation;
```
[the remark this answers is not in the document]
Comment (adwk67) [reply rust/crd/src/lib.rs]: Done.

**Comment by adwk67** [comment]
bors merge

**Pull request closed without merging**